src/main/java/sg/dex/crypto/Hash.java

```diff
@@ -123,4 +128,36 @@ public static String sha3_256String(String string) {
         return Hex.toString(sha3_256(string));
     }
 
+    /**
+     * This method is used to compute the complete hash computation based on input stream passed.
+     * A MessageDigest object starts out initialized.
+     * The data is processed through it using the update methods.
+     * At any point reset can be called to reset the digest.
+     * Once all the data to be updated has been updated, one of the digest
+     * methods should be called to complete the hash computation.
+     *
+     * @param inputStream for which the compute is to be performed
+     * @return compute hash value
+     */
+    public static String computeHashWithSHA3(InputStream inputStream) throws IOException {
+        String checksum;
+        if(null==inputStream){
+            throw new IOException("Asset content passed is  null");
+        }
+        try {
+
+            SHA3.DigestSHA3 md = new SHA3.Digest256();
+            //Using MessageDigest update() method to provide input
+            byte[] buffer = new byte[8192];
+            int numOfBytesRead;
+            while ((numOfBytesRead = inputStream.read(buffer)) > 0) {
+                md.update(buffer, 0, numOfBytesRead);
+            }
+            byte[] hash = md.digest();
+            checksum =  Hex.toString(hash);
+        } catch (IOException e) {
+            throw new IOException("Couldn't compute the hash  of partial content");
```

Comment:
What is "partial content"? Error message doesn't make sense. Just remove the whole try / catch block